sonic_y_cable/y_cable_base.py

```diff
@@ -0,0 +1,1130 @@
+        """
+
+        raise NotImplementedError
+
+    def enable_prbs_mode(self, target, mode_value, lane_mask, direction=PRBS_DIRECTION_BOTH):
+        """
+        This API configures and enables the PRBS mode/type depending upon the mode_value the user provides.
+        The mode_value configures the PRBS Type for generation and BER sensing on a per side basis.
+        Target is an integer for selecting which end of the Y cable we want to run PRBS on.
+        LaneMap specifies the lane configuration to run the PRBS on.
+        Note that this is a diagnostic mode command and must not run during normal traffic/switch operation
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+            target:
+                One of the following predefined constants, the target on which to enable the PRBS:
+                    EYE_PRBS_LOOPBACK_TARGET_LOCAL -> local side,
+                    EYE_PRBS_LOOPBACK_TARGET_TOR_A -> TOR A
+                    EYE_PRBS_LOOPBACK_TARGET_TOR_B -> TOR B
+                    EYE_PRBS_LOOPBACK_TARGET_NIC -> NIC
+            mode_value:
+                 an Integer, the mode/type for configuring the PRBS mode.
+
+            lane_mask:
+                 an Integer, representing the lane_mask to be run PRBS on
+                 0bit for lane 0, 1bit for lane1 and so on.
+                 for example 3 -> 0b'0011, means running on lane0 and lane1
+            direction:
+                One of the following predefined constants, the direction to run the PRBS:
+                    PRBS_DIRECTION_BOTH
+                    PRBS_DIRECTION_GENERATOR
+                    PRBS_DIRECTION_CHECKER
+
+        Returns:
+            a boolean, True if the enable is successful
+                     , False if the enable failed
+
+        """
+
+        raise NotImplementedError
+
+    def disable_prbs_mode(self, target, direction=PRBS_DIRECTION_BOTH):
+        """
+        This API disables the PRBS mode on the physical port.
+        The port on which this API is called for can be referred using self.port.
+
+        Args:
+            target:
+                One of the following predefined constants, the target on which to disable the PRBS:
+                    EYE_PRBS_LOOPBACK_TARGET_LOCAL -> local side,
+                    EYE_PRBS_LOOPBACK_TARGET_TOR_A -> TOR A
+                    EYE_PRBS_LOOPBACK_TARGET_TOR_B -> TOR B
+                    EYE_PRBS_LOOPBACK_TARGET_NIC -> NIC
+            direction:
+                One of the following predefined constants, the direction to run the PRBS:
+                    PRBS_DIRECTION_BOTH
+                    PRBS_DIRECTION_GENERATOR
+                    PRBS_DIRECTION_CHECKER
+
+        Returns:
+            a boolean, True if the disable is successful
+                     , False if the disable failed
+        """
+
+        raise NotImplementedError
+
+    def enable_loopback_mode(self, target, mode=NEAR_END_LOOPBACK, lane_mask):
```

Comment:
```suggestion
    def enable_loopback_mode(self, target, lane_mask, mode=LOOPBACK_MODE_NEAR_END):
```
Suggest to swap the order of lane_mask and mode to fix 'non-default argument follows default argument' syntax error and correct the variable name with 'LOOPBACK_MODE_NEAR_END'

Reply:
fixed